python-lib/text_extraction/__init__.py

```diff
@@ -45,6 +45,7 @@ def extract_text_content(file_bytes, extension, with_pandoc):
                 temporary_job_folder = os.getcwd()
                 with tempfile.NamedTemporaryFile(dir=temporary_job_folder, suffix=".{}".format(extension)) as tmp:
                     tmp.write(file_bytes)
+                    tmp.seek(0)
```

Comment:
```suggestion
                    tmp.flush() # Flush tmp file so pypandoc can use it
```

seek() isn't guaranteed to flush in general. e.g  in CPython, seek() will only flush when the seeked offset is outside of the current python-known file buffer : https://github.com/python/cpython/blob/v3.12.2/Modules/_io/bufferedio.c#L1358 is guarded by https://github.com/python/cpython/blob/v3.12.2/Modules/_io/bufferedio.c#L1353

Reply:
TIL. Actually based myself on the python documentation for this one (https://docs.python.org/3/library/tempfile.html#examples).
Let met try this fix, and see if it still works

Reply:
https://github.com/dataiku/dss-plugin-tesseract-ocr/pull/78/commits/ef6a6a4f092ac75e436dcaa2f347625b5150df02

Reply:
I think it's because in the examples they want to read() the file, so they have to seek(0) first to start at the beginning. And it can all happen in the memory of the current python process.

But here, it's a matter of launching a subprocess (pypandoc) that will access the file, so the file needs to be flushed to the OS filesystem (which doesn't mean flushed to disk yet btw 😂 ) for that to happen.